index.js

```diff
@@ -0,0 +1,99 @@
+function take(arr, n) {
+  var newArr = [];
+  var amount = n;
+  if (n > arr.length) {
+    amount = arr.length;
+  }
+  for (var i = 0; i < amount; i++) {
+    newArr.push(arr[i]);
+  }
+  return newArr;
+}
+
+
+function skip(arr, n) {
+  var newArr = [];
+  if (n > arr.length) {
+    return newArr;
+  }
+  for (var i = n; i < arr.length; i++) {
+    newArr.push(arr[i]);
+  }
+  return newArr;
+}
+
+function map(arr, callback) {
+  var newArr = [];
+  for (var i = 0; i < arr.length; i++) {
+    newArr.push(callback(arr[i], i, arr));
+  }
+  return newArr;
+}
+
+function reduce(arr, callback, initialValue) {
+  var initial = initialValue;
+  var i = 0;
+  if (!initial) {
+    initial = arr[0];
+    i = 1;
+  }
+  for (i; i < arr.length; i++) {
+    var current = arr[i];
+    initial = callback(initial, current);
+  }
+  return initial;
+}
+
+function filter(arr, callback) {
+  var newArr = new Array();
+  for (var i = 0; i < arr.length; i++) {
+    var boolean = callback(arr[i], i, arr);
+    if (boolean) {
+      newArr.push(arr[i]);
+    }
+  }
+  return newArr;
+}
+
+function foreach(arr, callback) {
+  for (var i = 0; i < arr.length; i++) {
+    if (arr.hasOwnProperty(i)) {
```

Comment:
to avoid using  `arr.hasOwnProperty` this method just use:

`if(arr[i] !== undefined)`